pallets/rewards/src/lib.rs

```diff
@@ -0,0 +1,959 @@
+                        attester_info
+                    } else {
+                        log::warn!(
+                            "No attester info found for during rewards distribution {:?}",
+                            attester
+                        );
+                        continue
+                    };
+
+                let commission_reward = attester_info.commission.mul_ceil(reward_per_attester);
+
+                // Update the pending claims for the attester
+                Self::update_pending_claims(
+                    &attester,
+                    CircuitRole::Attester,
+                    commission_reward,
+                    BenefitSource::Inflation,
+                );
+                let remaining_reward = reward_per_attester.saturating_sub(commission_reward);
+
+                // Get the attester's nominators
+                let nominators = T::Attesters::read_nominations(&attester);
+
+                let total_nomination: BalanceOf<T> = nominators
+                    .iter()
+                    .map(|(_, balance)| *balance)
+                    .fold(BalanceOf::<T>::zero(), |acc, x| acc.saturating_add(x));
+
+                // Distribute the remaining reward to the nominators
+                for (nominator, nomination_balance) in nominators {
+                    let check_nominator_reward = remaining_reward
+                        .saturating_mul(nomination_balance)
+                        .checked_div(&total_nomination);
+                    match check_nominator_reward {
+                        Some(nominator_reward) => {
+                            Self::update_pending_claims(
+                                &nominator,
+                                CircuitRole::Staker,
+                                nominator_reward,
+                                BenefitSource::Inflation,
+                            );
+                        },
+                        None => {
+                            // If the nominator reward is zero, then we don't need to do anything
+                            // because the nominator's balance is zero
+                            log::error!(
+                                "Nominator reward is zero for nominator {:?} and attester {:?}",
+                                nominator,
+                                attester
+                            );
+                        },
+                    }
+                }
+            }
+
+            BalanceOf::<T>::from(total_attesters).saturating_mul(reward_per_attester)
+        }
+
+        pub fn distribute_collator_rewards(current_distribution: BalanceOf<T>) -> BalanceOf<T> {
+            let total_collators = Collators::<T>::iter().count() as u32;
+
+            if total_collators == 0 {
+                return Zero::zero()
+            }
+
+            let reward_per_collator = current_distribution / BalanceOf::<T>::from(total_collators);
```

Comment:
Let us reconsider the way we reward collators here. 

Looking at the collator rewards, IMO there are two main things to consider and ensure:
1. For the collators we want to optimize for raw collator count. The more, the better.
2. We want to make sure, collators and attestors are not competing, resulting in stake leaving attestors and moving to collators. The network doesn't really benefit from more stake in collators as its dead/non-slashable staking


With these two things in mind, I feel the current setup is not optimal. In the end, the APY between collators and attestors should balance to be roughly the same (considering hardware requirements for collators).

The reward factor difference between attestors and collators is x2.2 (1.1% vs 0.5%). Considering the hardware costs, let's roughly say 30% of the stake will be with collators. I feel like we are wasting stake that can be rather with attesters to secure the network. The stake of an attester doesn't provide any benefit to the network, so we shouldn't reward it (especially as it's competing with attesters). Instead, we could consider requiring a min amount of stake, but don't reward anything above. This would result in excess stake going to attesters, securing the network.

In other news:
I still think the share of collator rewards should be higher, given that it economically secures the value of all cross-chain txs, in the worst case, over an entire slashing period.